convert/harness/yaml/step.go

```diff
@@ -293,20 +434,83 @@ func (s *Step) UnmarshalJSON(data []byte) error {
 	}
 
 	switch s.Type {
+	case StepTypeCustomApproval:
+		s.Spec = new(StepCustomApproval)
+	case StepTypeAction:
+		s.Spec = new(StepAction)
 	case StepTypeRun:
 		s.Spec = new(StepRun)
+	case StepTypeBarrier:
+		s.Spec = new(StepBarrier)
+	case StepTypeQueue:
+		s.Spec = new(StepQueue)
 	case StepTypePlugin:
 		s.Spec = new(StepPlugin)
 	case StepTypeBuildAndPushDockerRegistry:
 		s.Spec = new(StepDocker)
 	case StepTypeS3Upload:
 		s.Spec = new(StepS3Upload)
 	case StepTypeShellScript:
-		s.Spec = new(StepScript)
+		s.Spec = new(StepShellScript)
 	case StepTypeHTTP:
 		s.Spec = new(StepHTTP)
+	case StepTypeHarnessApproval:
+		s.Spec = new(StepHarnessApproval)
+	case StepTypeK8sRollingDeploy:
+		s.Spec = new(StepK8sRollingDeploy)
+	case StepTypeK8sRollingRollback:
+		s.Spec = new(StepK8sRollingRollback)
+	case StepTypeK8sApply:
+		s.Spec = new(StepK8sApply)
+	case StepTypeK8sBGSwapServices:
+		s.Spec = new(StepK8sBGSwapServices)
+	case StepTypeK8sBlueGreenStageScaleDown:
+		s.Spec = new(StepK8sBlueGreenStageScaleDown)
+	case StepTypeK8sCanaryDelete:
+		s.Spec = new(StepK8sCanaryDelete)
+	case StepTypeK8sDelete:
+		s.Spec = new(StepK8sDelete)
+	case StepTypeK8sDiff:
+		s.Spec = new(StepK8sDiff)
+	case StepTypeK8sRollout:
+		s.Spec = new(StepK8sRollout)
+	case StepTypeK8sScale:
+		s.Spec = new(StepK8sScale)
+	case StepTypeK8sDryRun:
+		s.Spec = new(StepK8sDryRun)
+	case StepTypeK8sTrafficRouting:
+		s.Spec = new(StepK8sTrafficRouting)
+	case StepTypeK8sCanaryDeploy:
+		s.Spec = new(StepK8sCanaryDeploy)
+	case StepTypeK8sBlueGreenDeploy:
+		s.Spec = new(StepK8sBlueGreenDeploy)
+	case StepTypeJiraCreate:
+		s.Spec = new(StepJiraCreate)
+	case StepTypeJiraUpdate:
+		s.Spec = new(StepJiraUpdate)
+	case StepTypeJiraApproval:
+		s.Spec = new(StepJiraApproval)
+	case StepTypeServiceNowApproval:
+		s.Spec = new(StepServiceNowApproval)
+	case StepTypeHelmBGDeploy:
+		s.Spec = new(StepHelmBGDeploy)
+	case StepTypeHelmBlueGreenSwapStep:
+		s.Spec = new(StepHelmBlueGreenSwapStep)
+	case StepTypeHelmCanaryDeploy:
+		s.Spec = new(StepHelmCanaryDeploy)
+	case StepTypeHelmDelete:
+		s.Spec = new(StepHelmDelete)
+	case StepTypeHelmDeploy:
+		s.Spec = new(StepHelmDeploy)
+	case StepTypeHelmRollback:
+		s.Spec = new(StepHelmRollback)
+	case StepTypeWait:
+		s.Spec = new(StepWait)
+	case StepTypeEmail:
+		s.Spec = new(StepEmail)
 	default:
-		return fmt.Errorf("unknown step type %s", s.Type)
+		log.Printf("unknown step type while unmarshalling %s", s.Type)
+		return nil
```

Comment:
should return error instead of silent failing
```suggestion
		return fmt.Errorf("unknown step type %s", s.Type)
```